explorer-api/src/geo_ip/location.rs

```diff
@@ -78,20 +124,24 @@ impl GeoIp {
                 &address
             );
             let p = port.unwrap_or(FAKE_PORT);
-            let socket_addr = (address, p)
-                .to_socket_addrs()
-                .map_err(|e| {
-                    error!("Fail to resolve IP address from {}:{}: {}", &address, p, e);
-                    GeoIpError::NoValidIP
-                })?
-                .next()
-                .ok_or_else(|| {
-                    error!("Fail to resolve IP address from {}:{}", &address, p);
-                    GeoIpError::NoValidIP
-                })?;
-            let ip = socket_addr.ip();
-            debug!("Internal lookup succeed, resolved ip: {}", ip);
-            Ok(ip)
+            match (address, p).to_socket_addrs() {
+                Ok(mut addrs) => {
+                    if let Some(socket_addr) = addrs.next() {
```

Comment:
you'd need to test it (but where would we find a qualified QA person? 🤔 ), but I think since this is an iterator, it might always return a `None` at the very end and thus you will always be hiting the `else` branch in every lookup.

so you could just do something along the lines of

```rust
for socket_addr in addr {
  let ip = socket_addr.ip();
  // do the rest 
}
```